Makefile

```diff
@@ -1,20 +1,60 @@
-.PHONY: build-stacks build-buildpacks build-builder
+PACK_CMD?=pack
 
-all: build-stacks build-buildpacks build-builder
+VERSION_TAG := v0.1.0
```

Comment:
If I am understanding this correctly, if we push a new commit to this repository, say `v1.2.3`, this version tag will need to also be updated to the same tag?

What do you think about changing this to:
```
GIT_TAG := $(shell git tag --points-at HEAD)
VERSION_TAG := $(shell [ -z $(VTAG) ] && echo 'tip' || echo $(GIT_TAG) )
```
This would set the version tag to the version tag of the commit if it exists (always set in CI since this command is only run on pushes with a tag beginning with v), and set to "tip" otherwise (for local builds or manual builds of a commit without a tag.  Intentially not "latest" such that a development build is not used by accident in produciton).

This does add some complexity, but it eliminates the possibility of updating one version and not the other (probably pretty likely, considering my own forgetfulness.)

Reply:
I like it. Two concerns I have remaining.

1. This versions all buildpack/builders together at the same time
2. All of the stack images are moving tags

This design is based on the official buildpack samples repo and is not set in stone. So just want to get thoughts around that too.

Reply:
One option discussed in https://github.com/boson-project/buildpacks/issues/5 is to append the version to the existing tag, e.g. `quay.io/boson/faas-stack-run:nodejs-v1.0.0`. Again, this means that _all_ images are versioned together.

Reply:
> This versions all buildpack/builders together at the same time

If we want to version them separately (perhaps a good idea!)  Maybe we would instead:

1) Create separate version nubmers for each repo:
```
QUARKUS_VERSION_TAG
NODEJS_VERSION_TAG
GO_VERSION_TAG
```
2) Update the `.github/workflows/releases.yaml` to run on a different condition
```
on:
  push:
    branches:
      - master
```
3) do not add any version tags in the git commit history at all to avoid confusion.

> All of the stack images are moving tags

I may not understand what you mean here; if we derive the version tag from the git commit, thre would be one floating tag ("tip") and one fixed tag (the version number of the git commit that generated it).

This question may be moot, however, if you think individually versioning the buildpacks is a better solution.

Reply:
With this, we would still be versioning a bunch of images together - i.e. all of the Node.js images (builder, buildpack and stack) all at once. I'm OK with that, but just wanted to point it out.

> I may not understand what you mean here; if we derive the version tag from the git commit, thre would be one floating tag ("tip") and one fixed tag (the version number of the git commit that generated it).

Well, the `stack` images are all non-versioned at the moment. See: https://quay.io/repository/boson/faas-stack-run?tab=tags. In #5 I proposed that we append the runtime tag with a version - e.g. `quarkus-v0.1.0` instead of just `quarkus`. With the latter, the `quarkus` tag just floats.

In general, I think the stack images will change a lot less frequently than the others, so I'm not terribly concerned. But still...

I'll poke on this today.

Reply:
Ah, I understand now.